transform-plugins/src/main/java/io/cdap/plugin/Flatten.java

```diff
@@ -0,0 +1,393 @@
+                                                                      config.getLevelToLimitFlattening());
+    this.inputOutputMapping = handleDuplicateFieldName(inputOutputMapping, config.getPrefix());
+    this.outputSchema = generateOutputSchema(inputSchema, inputOutputMapping);
+  }
+
+
+  @Override
+  public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter) throws Exception {
+    if (outputSchema == null) {
+      List<OutputFieldInfo> inputOutputMapping = createOutputFieldsInfo(input.getSchema(), config.getFieldsToFlatten(),
+                                                                        config.getLevelToLimitFlattening());
+      this.inputOutputMapping = handleDuplicateFieldName(inputOutputMapping, config.getPrefix());
+      this.outputSchema = generateOutputSchema(input.getSchema(), inputOutputMapping);
+    }
+    StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+
+    for (Schema.Field field : outputSchema.getFields()) {
+      OutputFieldInfo outputFieldInfo = inputOutputMapping.get(field.getName());
+      Object value = outputFieldInfo.getValue(input);
+      builder.set(field.getName(), value);
+    }
+    emitter.emit(builder.build());
+  }
+
+  private List<OutputFieldInfo> createOutputFieldsInfo(Schema inputSchema, List<String> fieldsToFlatten,
+                                                       int levelToLimitFlattening) {
+    List<Schema.Field> fields = inputSchema.getFields();
+    List<OutputFieldInfo> mapping = new ArrayList<>();
+
+    if (fields == null || fields.isEmpty()) {
+      return new ArrayList<>();
+    }
+    for (Schema.Field field : fields) {
+      String name = field.getName();
+      if (fieldsToFlatten.contains(name)) {
+        mapping.addAll(flattenField(field, levelToLimitFlattening));
+      } else {
+        mapping.add(generateOutputFieldInfo(field));
+      }
+    }
+    return mapping;
+  }
+
+  private List<OutputFieldInfo> flattenField(Schema.Field field, int levelToLimitFlattening) {
+    Schema schema = field.getSchema();
+
+    if (isRecord(schema) && levelToLimitFlattening != 0) {
+      return generateOutputFieldInfoForRecord(field, levelToLimitFlattening);
+    }
+
+    ArrayList<OutputFieldInfo> maps = new ArrayList<>();
+    maps.add(generateOutputFieldInfo(field));
+    return maps;
+  }
+
+  private OutputFieldInfo generateOutputFieldInfo(Schema.Field field) {
+    return generateOutputFieldInfo(field, field.getSchema());
+  }
+
+  private OutputFieldInfo generateOutputFieldInfo(Schema.Field field, Schema fieldSchema) {
+    Node node = new Node();
+    node.fieldName = field.getName();
+    node.fieldSchema = fieldSchema;
+
+    OutputFieldInfo outputFieldInfo = new OutputFieldInfo();
+    outputFieldInfo.fieldName = field.getName();
+    outputFieldInfo.node = node;
+    return outputFieldInfo;
+  }
+
+  private List<OutputFieldInfo> generateOutputFieldInfoForRecord(Schema.Field field, int levelToLimitFlattening) {
```

Comment:
The method is using DFS to flatten the schema. could you add a javadoc with a nested schema example on how it will be flatten